### PR TITLE
feat: Add no-exclusive-tests rule

### PR DESCRIPTION
**Pull request by RoryH**
Adds rule similar to other test frameworks to identify exclusive tests.

https://mochajs.org/#exclusive-tests

### CODE REVIEW

**Comment by jennifer-shehane** [comment]
I like it.

File: lib/rules/no-exclusive-tests.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      description: 'Prevent use of exclusive tests (i.e. .only())',
+      category: 'Warning',
+      recommended: true,
+    },
+    fixable: null, // or 'code' or 'whitespace'
+    schema: [
+      // fill in your schema
+    ],
+    messages: {
+      exclusiveTest: 'No exclusive tests',
```
Comment (Bkucera) [inline lib/rules/no-exclusive-tests.js]: I would prefer 'Unexpected exclusive Cypress test'

**Comment by RoryH** [comment]
@Bkucera I had considered possibly re-using eslint-plugin-mocha somehow but this is my first time diving into eslint plugins 🤓 . How do you see it working without adding it as an extra dependency, would it be transitive dependency of eslint-plugin-cypress?

**Comment by Bkucera** [comment]
@RoryH we add it as a dependency, but we require the rules directly, like they do in this file https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/index.js

**Comment by RoryH** [comment]
I like it... If you guys are happy I'll close this PR and open a new one pulling in the rules from `eslint-plugin-mocha` module 👍

**Comment by Bkucera** [comment]
@RoryH sounds great

**Pull request closed without merging**